sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_client.hpp

```diff
@@ -111,10 +114,39 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
           context,
           Azure::Core::Http::HttpMethod::Post,
           Details::KeyRequestParameters(keyType, options),
-          [name](Azure::Core::Http::RawResponse const& rawResponse) {
+          [&name](Azure::Core::Http::RawResponse const& rawResponse) {
             return Details::KeyVaultKeyDeserialize(name, rawResponse);
           },
           {Details::KeysPath, name, "create"});
     }
+
+    /**
+     * @brief Deletes a key of any type from storage in Azure Key Vault.
+     *
+     * @remark The delete key operation cannot be used to remove individual versions of a key. This
+     * operation removes the cryptographic material associated with the key, which means the key is
+     * not usable for Sign/Verify, Wrap/Unwrap or Encrypt/Decrypt operations. This operation
+     * requires the keys/delete permission.
+     *
+     * @param name The name of the key.
+     * @param context A cancellation token controlling the request lifetime.
+     * @return A #DeleteKeyOperation to wait on this long-running operation. If the key is soft
+     * delete-enabled, you only need to wait for the operation to complete if you need to recover or
+     * purge the key; otherwise, the key is deleted automatically on purge schedule.
+     */
+    Azure::Security::KeyVault::Keys::DeleteKeyOperation StartDeleteKey(
```

Comment:
We use "Start" in .NET only to avoid looking like an old-style async "Begin" method. Other languages use "begin". If "Start" is what C/C++ is going with, cool. Just pointing out that .NET is different here than the existing language SDKs.

Reply:
We are also using `Start` as convention in C++.
See this LRO from storage:

```cpp
Azure::Core::Response<Models::StartCopyBlobResult> StartCopyFromUri(
        const std::string& sourceUri,
        const StartCopyBlobFromUriOptions& options = StartCopyBlobFromUriOptions(),
        const Azure::Core::Context& context = Azure::Core::Context()) const;
```